ClientTests/FeatureFlagManagerTests.swift

```diff
@@ -0,0 +1,228 @@
+            XCTAssertTrue(featureFlags.isFeatureEnabled(.historyGroups, checking: .buildOnly))
+            XCTAssertTrue(featureFlags.isFeatureEnabled(.historyGroups, checking: .userOnly))
+            XCTAssertTrue(featureFlags.isFeatureEnabled(.inactiveTabs, checking: .buildOnly))
+            XCTAssertTrue(featureFlags.isFeatureEnabled(.inactiveTabs, checking: .userOnly))
+            XCTAssertTrue(featureFlags.isFeatureEnabled(.jumpBackIn, checking: .buildOnly))
+            XCTAssertTrue(featureFlags.isFeatureEnabled(.jumpBackIn, checking: .userOnly))
+            XCTAssertTrue(featureFlags.isFeatureEnabled(.pocket, checking: .buildOnly))
+            XCTAssertTrue(featureFlags.isFeatureEnabled(.pocket, checking: .userOnly))
+            XCTAssertTrue(featureFlags.isFeatureEnabled(.pullToRefresh, checking: .buildOnly))
+            XCTAssertTrue(featureFlags.isFeatureEnabled(.pullToRefresh, checking: .userOnly))
+            XCTAssertTrue(featureFlags.isFeatureEnabled(.recentlySaved, checking: .buildOnly))
+            XCTAssertTrue(featureFlags.isFeatureEnabled(.recentlySaved, checking: .userOnly))
+            XCTAssertTrue(featureFlags.isFeatureEnabled(.reportSiteIssue, checking: .buildOnly))
+            XCTAssertTrue(featureFlags.isFeatureEnabled(.reportSiteIssue, checking: .userOnly))
+            XCTAssertTrue(featureFlags.isFeatureEnabled(.shakeToRestore, checking: .buildOnly))
+            XCTAssertTrue(featureFlags.isFeatureEnabled(.shakeToRestore, checking: .userOnly))
+            XCTAssertTrue(featureFlags.isFeatureEnabled(.sponsoredTiles, checking: .buildOnly))
+            XCTAssertTrue(featureFlags.isFeatureEnabled(.sponsoredTiles, checking: .userOnly))
+            XCTAssertTrue(featureFlags.isFeatureEnabled(.startAtHome, checking: .buildOnly))
+            XCTAssertTrue(featureFlags.isFeatureEnabled(.startAtHome, checking: .userOnly))
+            XCTAssertTrue(featureFlags.isFeatureEnabled(.tabTrayGroups, checking: .buildOnly))
+            XCTAssertTrue(featureFlags.isFeatureEnabled(.tabTrayGroups, checking: .userOnly))
+            XCTAssertTrue(featureFlags.isFeatureEnabled(.topSites, checking: .buildOnly))
+            XCTAssertTrue(featureFlags.isFeatureEnabled(.topSites, checking: .userOnly))
+            XCTAssertTrue(featureFlags.isFeatureEnabled(.wallpapers, checking: .buildOnly))
+
+        #else
+            XCTFail("We are in an unknown build type. What's happening!? How did things go so wrong!!")
+        #endif
+    }
+
+    func testDefaultNimbusCustomFlags() {
+        XCTAssertEqual(featureFlags.getCustomState(for: .searchBarPosition), SearchBarPosition.bottom)
+        XCTAssertEqual(featureFlags.getCustomState(for: .startAtHome), StartAtHomeSetting.afterFourHours)
+    }
+
+    // Changing the prefs manually, to make sure settings are respected through
+    // the FFMs interface
+    func testFFMRespectsProfileChangesForBoolSettings() {
```

Comment:
FFM stands for?

Reply:
FeatureFlagsManager

Reply:
Just a nit, I'm not a fan of acronyms -> `testManagerRespectsProfileChangesForBoolSettings` ?

Reply:
sure